src/cpp/session/modules/SessionRCompletions.R

```diff
@@ -1526,16 +1526,30 @@ assign(x = ".rs.acCompletionTypes",
    
    if (isAt)
    {
-      if (isS4(object))
+      if (exists(".AtNames"))
       {
-         tryCatch({
+         .rs.tryCatch({
+            code <- substitute(
+               utils::.AtNames(object, pattern = ""),
+               list(object = object)
+            )
+            allNames <- eval(code, envir = globalenv())
+            names <- .rs.selectFuzzyMatches(allNames, token)
+            type <- attr(names, "types")
+         })
+      }
+      else if (isS4(object))
```

Comment:
Actually on second thought I don't get why this is an else if? So we only do the s4 slot completions on older R versions where AtNames is not defined?

Reply:
Oh I see now that S4 objects also use the .AtNames so this works for standard S4 object definitions too